File: python/sorting/selection_sort.py
def selection_sort(list):
    sorted_list = []
    print("%-25s %-25s" % (list, sorted_list))
    for i in range(0,len(list)):
        moving_index = index_min(list)
        sorted_list.append(list.pop(moving_index))
        print("%-25s %-25s" % (list, sorted_list))
    return sorted_list 

def index_min(list):
    min_index = 0 
    for i in range(1, len(list)):
        if list[i] < list[min_index]:
            min_index = i
    return min_index

alist = [11,2,44,15,6,9,7]
verify = selection_sort(alist)
print(verify)

File: javascript/sorting/selectionSort.js
function selectionSort(arr){
    sortedList = []
    for(i=0; i<arr.length; i++){
        movingIndex = indexMin(arr)
        sortedList.push(arr[movingIndex])
        arr.pop(movingIndex)

        console.log("poped array",arr)
        console.log(sortedList)
    }
    return sortedList
}

function indexMin(arr){
    let minIndex = 0
    for(i=1; i<arr.length; i++){
        if(arr[i] < arr[minIndex]){
            minIndex = i
        }
    } 
    return minIndex
}

arr = [11,2,44,15,6,9,7]
verify = selectionSort(arr)
// console.log(verify)

// for (i=0; i<5; i++){
//     console.log("this is the number", i)
// }